### DISCUSSION TITLE
Plans for Adobe Scout

### DISCUSSION THREAD

**Discussion by PippoApps** [airsdk/Adobe-Runtime-Support#2001] [Ideas]
Hello there,
I was wondering if you plan to ever take on Adobe Scout.
I use it often and find it quite valuable. Still last release was some years go and it has some annoying bugs :)
Or would you suggest an alternative?
Cheers, and thank you for the amazing work with AIR SDK
F

**Comment by ajwfrost** [comment]
Hi

We have got the source code for Scout now, it's a relatively recent addition as we also needed to get some associated permissions... but we are now able to modify and deploy it ourselves, with a limitation that Adobe imposed that it should always be free :-)

I'm wondering whether we should have a separate 'project' on github for Scout, rather than raise issues under the "Adobe-Runtime-Support" area. Or they could just be tagged with a category for Scout .. 

In terms of maintaining Scout .. the initial reason we got it was because the Android helper app for Scout that allows you to configure the connections was using a mechanism that no longer works on recent Android API levels. So we've got our own deployment of the Scout mobile configuration tool on the Play Store now, for people who don't just have the settings file included in their APKs..

We looked briefly at building this for the desktop platforms: it needs very old tools so we've not actually got this building yet, it didn't seem to be a particularly high priority. But we can revisit that at some point and see if we can get it building, and then start addressing some of the bugs/issues.

thanks

**Reply by marchbold** [reply]
@ajwfrost Definitely think it's worth setting up a separate repo to maintain scout. I think it's worth it to keep the issues / discussions clear and on topic.

Let me know if you want me to do this

**Reply by ajwfrost** [reply]
> Let me know if you want me to do this

@marchbold I was just thinking, I really ought to learn how to do this sort of thing! So have just done it (I hope!!)
https://github.com/airsdk/Adobe-Scout

**Reply by marchbold** [reply]
Ha ha, nice :)

**Comment by 2jfw** [comment]
@ajwfrost Would be nice, for long-term testing, if you could remove the memory limitation of Scout, which will currently stop working when one or many sessions have exceeded ~2 GB of RAM

**Comment by PippoApps** [comment]
Dear @ajwfrost,
thank you for sharing interesting details.
As a wish for the future, I would love to be able to use Scout to read real-time trace actions: ATM the only way is by selecting the entire timeline, but each new trace entry, makes trace content scroll back to the top so it is basically useless.

I am resorting to a hidden WebView that routes trace messages to https://console.re, at least I can see live in browser all debug messages also on client debug releases.

Cheers mate, looking forward to new versions of scout.
I cannot imagine the headache caused by compiling a package from a codebase updated abt 10 years ago...

**Comment by shaucky** [comment]
Does anyone else remember Adobe Scout in their hearts? Or is it too difficult to restore updates to it after a long-term review?

**Reply by ajwfrost** [reply]
Feel free to add comments/issues: https://github.com/airsdk/Adobe-Scout/issues

We actually got this building for Windows recently, it's fairly old code and pretty unfamiliar for us, but we may be able to do something with it...

**Comment by busitech** [comment]
Flex Builder is the application that you need the source code for.  Flex Builder was much better at assisting the developer in removing memory leaks than Scout, because Flex Builder Profiler could:
- capture as many Memory Snapshots as desired for later comparison
- generate a list of Loitering Objects between any two Memory Snapshots
- sort objects by Class, Package, Retained Memory, Shallow Memory
- show the Object References graph starting with the last object held all the way through the GC Root, along with naming the specific property holding the reference to the next level
- clearly identify all GC Roots responsible for the objects retained
- record and display generated Object Allocation Stack Traces for any object, just by double clicking in the list of Loitering Objects or Object References
- see a list of Live Objects while the program is running
- save and restore Profiling Data
- display a line chart of current and peak memory usage over time
- display the properties of each retained object, and values for each property
- jump from any frame in the stack trace directly to the source code

Customers who OWN copies of Flash Builder, but need to reinstall the application, are completely unable to activate the software they've paid for, which is still the best tool for certain tasks (see the above list).

Would there be any hope that Harman could acquire the Flash Builder source code, and start assisting customers with activating their existing licenses?

Otherwise, the above list would be the minimum required to ADD to Scout before it could be on par with Flash Builder.

**Reply by hardcoremore** [reply]
Hi @ajwfrost,

Are there any news about the Scout? It would be great to have it on Mac also and to have function call count.  Flash Builder had function call count and was very useful.

**Reply by ajwfrost** [reply]
Scout is available on macOS and Windows: https://airsdk.dev/docs/tools/development/scout/getting-started#install-scout
We've not done a lot with it, but feel free to add requests to https://github.com/airsdk/Adobe-Scout/issues
Sadly we're not able to get the Flash Builder sources or resurrect that. I quite liked the design view in the older Flex Builder versions..

**Reply by hardcoremore** [reply]
Hi @ajwfrost,

I am trying to install Scout on Mac Mini M4. Installation goes without problem but when I try to open Scout I got this

<img width="253" alt="image" src="https://github.com/user-attachments/assets/24e119c5-f979-43c6-9a5f-79af1b1e326d" />

**Reply by ajwfrost** [reply]
When I download the DMG file, I can open it and run the installer (PKG) at which point it warns me that the certificate has expired (which I would think is okay seeing as it was valid when the signing happened, and it's not been revoked..)

I don't seem to then have a problem with it. Although I'm only on M2, I'm on macOS 15.2 so I'd have thought the security would be the same...

Can you double-check:
```
ls -la /Applications/Adobe\ Scout\ CC.app
```
shouldn't have any extended attributes (the `@` symbol) - if it does, this may be the issue, you can list them with `xattr folder` and remove with `sudo xattr -r -d attribute_name folder`

Plus:
```
codesign -d -vvv /Applications/Adobe\ Scout\ CC.app
```
should show Adobe's Developer ID Application certificate with code directory version 20100...? Would be good to also check the 'verify' of the signature
```
codesign -v -vvv /Applications/Adobe\ Scout\ CC.app
```

My guess from that message is that there's a quarantine flag set, but not 100% sure..

thanks

**Reply by hardcoremore** [reply]
I got this:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/4a42c155-ecec-4fd0-8a93-a530af4a9476" />


And this


<img width="722" alt="image" src="https://github.com/user-attachments/assets/6d063c54-9ec3-4a8d-b917-6521f41d9966" />

And this

<img width="677" alt="image" src="https://github.com/user-attachments/assets/30f7136b-7277-4bc1-9820-3376cfe675e8" />


For all 3 commands.

**Reply by ajwfrost** [reply]
Hmmm.. well, that really sounds like it should not be possible for anyone to use it on recent macOS versions! Version 1 signature is not meant to have been accepted for quite a while. And odd that I don't see the same output...

There's some troubleshooting suggestions here:
https://forums.developer.apple.com/forums/thread/706379

But ultimately I suspect we will need to rebuild and re-sign it all, in order to make it a bit more future-proof... let me see if someone can take a look..

**Comment by vync79** [comment]
Hi everyone, 
is there any plan to revamp Adobe Scout as a VS Code profiler extension for example ?
